src/ape/api/address.py

```diff
@@ -54,6 +54,18 @@ def address(self) -> AddressType:
             :class:`~ape.types.AddressType`
         """
 
+    def __eq__(self, __o: object) -> bool:
+        """
+        Compares AddressAPI/str objs
+
+        Returns:
+            bool: comparison result
+        """
+
+        if isinstance(__o, str):
+            return self.address == __o
```

Comment:
```suggestion
            return super().__eq__(self.address, __o)
```

Would this make sense?

Reply:
Because we're overriding it by comparing it to an item that isn't the same type we can't use the super method, it results in this:
```python
TypeError: super(type, obj): obj must be an instance or subtype of type
```